### ISSUE TITLE
Problem with prediction of non-linear SVM

### ISSUE THREAD

**Issue by FelipeBerrios** [nfmcclure/tensorflow_cookbook#102]
When I try to predict for new entries using the test set, it seems to depend on how much data is passed to the prediction operation. For example, if you only pass one entry data, it always predicts the same class, even though it is erroneous. As you pass more test data, you begin to predict different values.
My question is whether this behavior is normal ?. In addition the results seem to depend on the random values ​​rand_x and rand_y. Thanks for the help.

**Comment by FelipeBerrios** [comment]
I realized that if only one test sample is passed to it, the prediction always returns a vector with zero values ​​since it subtracts prediction_output with itself, so tf.argmax always returns the index zero.

`prediction_output = tf.matmul(tf.multiply(y_target,b), pred_kernel)`
`prediction = tf.arg_max(prediction_output-tf.expand_dims(tf.reduce_mean(prediction_output,1), 1), 0)`

Then by passing only one element to prediction_grid, the subtraction returns:

`[[ 0.]`
 `[ 0.]`
 `[ 0.]]`

And prediction takes the value `[0]`

**Comment by nfmcclure** [comment]
Hi @FelipeBerrios ,  This is interesting, thanks for finding this-  and sorry for the late reply. I'm just getting around to triaging all the issues in preparation for a v2 of the code.

I'll probably get around to chapter 4 in the next month, so I'll look into this then. Thanks again.
